fetcher/fedora.go

```diff
@@ -0,0 +1,293 @@
+package fetcher
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/inconshreveable/log15"
+	"github.com/vulsio/goval-dictionary/util"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v2"
+)
+
+// FetchFedora fetch OVAL from Fedora
+func FetchUpdateInfosFedora(versions []string) (fedoraUpdatesPerVersion, error) {
+	reqs, moduleReqs := newFedoraFetchRequests(versions)
+	results, err := fetchEverythingFedora(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("fetchEverythingFedora: %w", err)
+	}
+
+	moduleResults, err := fetchModulesFedora(moduleReqs)
+	if err != nil {
+		return nil, xerrors.Errorf("fetchModulesFedora: %w", err)
+	}
+	results.merge(&moduleResults)
+
+	return results, nil
+}
+
+func newFedoraFetchRequests(target []string) (reqs, moduleReqs []fetchRequest) {
+	const href = "https://dl.fedoraproject.org/pub/fedora/linux/updates/%s/Everything/x86_64/repodata/repomd.xml"
+	const moduleHref = "https://dl.fedoraproject.org/pub/fedora/linux/updates/%s/Modular/x86_64/repodata/repomd.xml"
```

Comment:
Everything's updateinfo lists x86_64, aarch64, etc., but Modular only lists x86_64 for x86_64.
Can you support Modular's aarch64?